src/map/Map.cpp

```diff
@@ -0,0 +1,239 @@
+/*------------------------------------------------------------------------------
+* Source: Map.cpp
+*
+* Functions:
+*       Map(string file);
+*       int loadFileData();
+*       void genWalls(const struct MapPoint wstart[MAX_WALLS]);
+*       string getFile();
+*       void setFile(string f);
+*       void mapLoadToGame();
+*       void printData();
+*
+* Date: March 23, 2017
+*
+* Revisions:
+* Edited By : Name 2016/JAN/25 - Description
+*
+* Designer: Jordan Lin
+*
+* Author: Jordan Lin
+*
+* Notes:
+* Map Class functions used for reading in a map file and loading the map into the game.
+------------------------------------------------------------------------------*/
+#include "Map.h"
+
+using namespace std;
+
+/**
+ * Date: Mar. 23, 2017
+ * Author: Jordan Lin
+ * Function Interface: Map(string file)
+ * Parameters:
+ *      file - File name, must be a CSV File
+ * Description:
+ * Map Object Constructor, creates a map object using the map selected file name.
+ */
+Map::Map(string file):fname(file) {
+    for(int i = 0; i < M_HEIGHT; i++) {
+        for(int j = 0; j < M_WIDTH; j++) {
+            AIMap[i][j] = 0;
+            mapdata[i][j] = 0;
+        }
+    }
+}
+
+/**
+ * Date: Mar. 23, 2017
+ * Author: Jordan Lin
+ * Function Interface: int loadFileData()
+ * Description:
+ * Read's the Map CSV file map data with the appropriate tile IDs
+ */
+int Map::loadFileData() {
+    // Open file stream for selected map file
+    ifstream file;
+    file.open(fname.c_str());
+
+    if (file.fail()) {
+        printf("failed to open file\n");
+        return 0;
+    }
+
+    // Array of wallstart positions
+    struct MapPoint wallStart[MAX_WALLS];
+    // Indices
+    int j = 0;
+    int i = 0;
+
+    // Character Variable
+    char ch;
+    while (file >> noskipws >> ch) {
```

Comment:
```
std::ifstream t("file.txt");
t.seekg(0, std::ios::end);
size_t size = t.tellg();
std::string buffer(size, ' ');
t.seekg(0);
t.read(&buffer[0], size); 
```

Use this for reading the file into a string
Then iterate over the string to process the file
Doing a while loop on a filestream like this reeks of slowness, and while it's not a huge issue since our files won't be several megs, there's no reason to write it slowly